README.md

```diff
@@ -1,2 +1,319 @@
+    - TODO: decide on list of themes to natively support and create and style them.
+- `position`: one of `"top-left"`, `"top-center"`, `"top-right"`, `"middle-left"`, `"middle-center"`, `"middle-right"`, `"bottom-left"`, `"bottom-center"`, or `"bottom-right"`. The default (if the attribute is omitted or set to an invalid value) is `"bottom-right"`.
+    - TODO: this seems stylistic; should it be controlled via CSS instead?
+    - TODO: these are physical positions; should we use [logical ones](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Logical_Properties/Basic_concepts) instead?
+- `closebutton`: a boolean attribute, determining whether an explicit close button is shown. By default toasts do not have a close button. TODO
+
+All attributes will be reflected as properties on the element's JavaScript interface.
+For example:
+
+```js
+const toast = document.createElement('std-toast');
+console.log(toast.open); // false
+```
+
+#### Contents
+
+The `<std-toast>` element can be used as a generic container,
+but will work best (in terms of default styling and events) if the developer conforms to the following content model:
+
+- The first child node,
+  typically either a text node or a container like `<p>`,
+  provides a message.
+- If more than one child node is present,
+  then the last child element,
+  which can be either an `<a>` or a `<button>`,
+  provides a call to action.
+
+TODO: what about title or icon? They should potentially also be accomodated, in a similar fashion.
+
+Thus, the following would all work well out of the box:
+
+```html
+<std-toast>Hello world!</std-toast>
+
+<std-toast><p>Hello world!</p></std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <button>Click me!</button>
+</std-toast>
+
+<std-toast>
+  <p>Hello world!</p>
+  <a href="https://example.com/" target="_blank">Click me!</button>
+</std-toast>
+```
+
+These can equivalently be created via JavaScript:
+
+```js
+// TODO: write this once we figure out the JS API for supplying HTML messages,
+// and for supplying customizable actions
+```
+
+More complex toasts,
+that don't fit the above content model,
+would require more custom handling on the part of the developer:
+
+```html
+<std-toast>
+  <p>Hello world!</p>
+  <p>A second paragraph?!</p>
+
+  <button>Action 1</button>
+  <button>Action 2</button>
+</std-toast>
+```
+
+Such an unusual toast would still integrate with other toasts in terms of stacking and positioning behavior,
+and some of the default styles that are inherited may be useful.
+But the page author will need to handle the action clicks themselves (instead of using the `"actionclick"` event),
+and will need to add additional styling to handle the extra contents.
+
+TODO: when we have a prototype, link to/show an example of this in action.
+
+#### Methods
+
+The `.show(options)` function shows a toast element,
```

Comment:
no `hide`/`dismiss` method? If that's intentional, we should explain our reasons.

Reply:
Good question. @jackbsteinberg, did this appear much in the libraries you surveyed?

Reply:
Yeah libraries usually had a `close` or `dismiss` function available, that's something I could add directly to the research if it would be a value add to have there.

Reply:
I'm adding this to the table of comparison in the study group.

Reply:
I added this to the API as well